modules/flyte/groups.tf

```diff
@@ -0,0 +1,16 @@
+resource "azuread_group" "flyte_metadata_access" {
+  display_name     = "Flyte metadata access"
+  security_enabled = true
+  members = [
+    azurerm_user_assigned_identity.flyte_controlplane.id,
+  ]
+}
+
+resource "azuread_group" "flyte_data_access" {
+  display_name     = "Flyte data access"
+  security_enabled = true
+  members = [
+    azurerm_user_assigned_identity.flyte_controlplane.id,
+    azurerm_user_assigned_identity.flyte_dataplane.id,
+  ]
+}
```

Comment:
I think we need to flip this: The controlplane role should only have access to the meta bucket/container, but the dataplane role should have access to both